### ISSUE TITLE
valid-title rule: Please allow function-typed variables and aliases as valid describe titles

### ISSUE THREAD

**Issue by y-hsgw** [vitest-dev/eslint-plugin-vitest#707]
### Summary
Currently, the `vitest/valid-title `rule reports an error if a function-typed variable (such as a variable typed with a type alias like `type Func = (params: Params) => void`) is used as the first argument to describe or test.
However, this lint rule only reports an error when a function-typed variable using a type alias is passed as the title. If a named function (declared without a type alias) is used directly as the title, no error is reported. 

### Example

```ts
type Func = (props: object) => void
const func: Func = (props) => console.log(props)
describe(func, () => {
  // ...
});
```

The above code works as expected in Vitest, but the lint rule reports:
Test title must be a string, a function or class name (vitest/valid-title)

### Expected behavior
The rule should recognize variables or constants that are typed as functions (including type aliases like Func above, React.FC, etc.) as valid titles for describe and test.
I think it would be more convenient if any variable or constant that is statically typed as a function were accepted (not just named function declarations).

### Actual behavior
The rule currently only accepts named functions or classes, but not variables typed as functions or function aliases.

Thank you for your consideration!

**Comment by veritem** [comment]
Thanks for bringing this to my attention.

it looks like this if condition should be extended to check typed variables as well. https://github.com/vitest-dev/eslint-plugin-vitest/blob/2818d59ba1a5141af6e7165c162d90da72ab8037/src/rules/valid-title.ts#L208
